### PR TITLE
Added Javadoc. Removed unnecessary whitespaces.

### PR DESCRIPTION
**Pull request by Shaswata79**

### CODE REVIEW

**Review by T-Syme — APPROVED** [review]

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/AdminController.java [hunk 1]
```diff
@@ -127,6 +127,13 @@
 	}
 
 
+	/**
+	 * POST request to change password
+	 * @param email
+	 * @param newPassword
+	 * @param token
+	 * @return
```
Comment (sandyl289) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/controller/AdminController.java]:
```suggestion
	 * @return an admin dto
```

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/AdminRepository.java [hunk 2]
```diff
@@ -8,12 +8,29 @@
 
 public interface AdminRepository extends CrudRepository<Admin, String> {
 	
+	/**
+	 * Find an admin by email
+	 * @param email
+	 * @return
```
Comment (sandyl289) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/AdminRepository.java]:
```suggestion
	 * @return admin (Admin)
```

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/AdminRepository.java [hunk 3]
```diff
@@ -8,12 +8,29 @@
...
+	/**
+	 * Delete the admin with the provided email
+	 * @param email
+	 */
 	void deleteAdminByEmail(String email);
 
+	/**
+	 * Find an admin by token
+	 * @param token
+	 * @return
```
Comment (sandyl289) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/AdminRepository.java]:
```suggestion
	 * @return admin (Admin)
```

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/AppointmentRepository.java [hunk 4]
```diff
@@ -10,17 +10,41 @@
 
 public interface AppointmentRepository extends CrudRepository<Appointment, Long>{
 
-	//List of appointments the customer has booked
+	/**
+	 * Find all the appointments a customer has booked
+	 * @param customer
+	 * @return
```
Comment (sandyl289) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/AppointmentRepository.java]:
```suggestion
	 * @return list of customer's appointments (List<Appointment>)
```

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/AppointmentRepository.java [hunk 5]
```diff
@@ -10,17 +10,41 @@
...
+	 */
 	List<Appointment> findAppointmentByCustomer(Customer customer);
 
+	/**
+	 * Find all the appointments booked for a given service
+	 * @param service
+	 * @return
```
Comment (sandyl289) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/AppointmentRepository.java]:
```suggestion
	 * @return List of appointments (List<Appointment>)
```

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/AppointmentRepository.java [hunk 6]
```diff
@@ -10,17 +10,41 @@
...
+	 */
 	List<Appointment> findByService(Service service);
 
+	/**
+	 * Find appointment by ID
+	 * @param appointmentID
+	 * @return
```
Comment (sandyl289) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/AppointmentRepository.java]:
```suggestion
	 * @return appointment with matching ID (Appointment)
```

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/BusinessRepository.java [hunk 7]
```diff
@@ -7,7 +7,15 @@
 
 public interface BusinessRepository extends CrudRepository<Business, Long> {
 	
+	/**
+	 * Find a business by ID
+	 * @param businessID
+	 * @return
```
Comment (sandyl289) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/BusinessRepository.java]:
```suggestion
	 * @return business (Business)
```

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/CustomerRepository.java [hunk 8]
```diff
@@ -8,13 +8,29 @@
 
 public interface CustomerRepository extends CrudRepository<Customer, String> {
 
+	/**
+	 * Find a customer by email
+	 * @param email
+	 * @return
```
Comment (sandyl289) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/CustomerRepository.java]:
```suggestion
	 * @return customer (Customer)
```

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/ReminderRepository.java [hunk 9]
```diff
@@ -9,7 +9,21 @@
 
 public interface ReminderRepository extends CrudRepository<Reminder, Long> {
 
+	/**
+	 * Find all the reminders for a certain customer
+	 * @param customer
+	 * @return
```
Comment (sandyl289) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/ReminderRepository.java]:
```suggestion
	 * @return list of a customer's reminders (List<Reminder>)
```

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/ServiceRepository.java [hunk 10]
```diff
@@ -7,8 +7,16 @@
 
 public interface ServiceRepository extends CrudRepository<Service, String> {
 
+	/**
+	 * Find a service by name
+	 * @param name
+	 * @return
```
Comment (sandyl289) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/ServiceRepository.java]:
```suggestion
	 * @return service (Service)
```

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/service/AuthenticationService.java [hunk 11]
```diff
@@ -98,6 +120,13 @@ private String authenticateAdmin(String email, String password) throws Authentic
         }
     }
 
+    /**
+     * Method to log in a customer
+     * @param email
+     * @param password
+     * @return
```
Comment (sandyl289) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/service/AuthenticationService.java]:
```suggestion
     * @return customer's token or error message (String)
```

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/service/AuthenticationService.java [hunk 12]
```diff
@@ -88,6 +103,13 @@ private String authenticateTechnician(String email, String password) throws Auth
         }
     }
 
+    /**
+     * Method to log in an admin
+     * @param email
+     * @param password
+     * @return
```
Comment (sandyl289) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/service/AuthenticationService.java]:
```suggestion
     * @return admin's token or error message (String)
```

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/service/AuthenticationService.java [hunk 13]
```diff
@@ -78,6 +84,15 @@ public Customer validateCustomerToken(String token) {
         return customerRepository.findCustomerByToken(token);
     }
 
+    
+    
+    /**
+     * Method to log in a technician
+     * @param email
+     * @param password
+     * @return
```
Comment (sandyl289) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/service/AuthenticationService.java]:
```suggestion
     * @return technician's token or error message (String)
```

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/TechnicianRepository.java [hunk 14]
```diff
@@ -10,11 +10,28 @@
...
+	/**
+	 * Delete the technician with the given email
+	 * @param email
+	 */
 	void deleteTechnicianByEmail(String email);
 
+	/**
+	 * Find a technician by token
+	 * @param token
+	 * @return
```
Comment (sandyl289) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/TechnicianRepository.java]:
```suggestion
	 * @return technician found (Technician)
```

File: RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/TechnicianRepository.java [hunk 15]
```diff
@@ -10,11 +10,28 @@
 
 public interface TechnicianRepository extends CrudRepository<Technician, String>{
 	
+	/**
+	 * Find a technician by email
+	 * @param email
+	 * @return
```
Comment (sandyl289) [inline RepairShop-Backend/src/main/java/ca/mcgill/ecse321/repairshop/repository/TechnicianRepository.java]:
```suggestion
	 * @return technician found (Technician)
```

**Review by sandyl289 — APPROVED** [review]
Thanks Shaswata!! 👍🏼 💯 

I left some suggestions for some `@return`.

**Review by zhannaklimanova — APPROVED** [review]

**Pull request merged**